woocommerce/woocommerce-android
# [Woo POS][non-simple products] Integrate "include_types" parameter for product endpoint

### NOTE: This PR depends on [this fluxc PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3117). Please review and merge that before merging this one. Also, please ensure to update the fluxc version after merging the fluxc PR.

⚠️ Don't merge until the base branch is not merged into trunk.

Closes: #13061 


### Description

This PR integrates include_types parameter for the product endpoint. This parameter helps especially in POS to include only product types that are compatible. Right now, we have enabled just `Simple` product types for POS.

With this integration, `include_types` takes precedence over the type param when both are present.

Here's the backend PR for more context: https://github.com/woocommerce/woocommerce/pull/52994

This PR also removes non-simple products feature flag as it is no longer required since we can control via the API the type of product we display in POS.



### Testing information

1. Ensure you have installed the [latest Woocommerce nightly build](https://github.com/woocommerce/woocommerce/releases) in your store
2. Navigate to the products tab and ensure you see all the products
3. Open network inspection tab in Android studio and ensure the network call has `include_types=`
4. Navigate to the POS (More menu -> POS)
5. Ensure you see only simple products
6. Open network inspection tab in Android studio and ensure the network call has `include_types=simple`

### The tests that have been performed

Ensured only simple products are displayed on POS mode and all products in store management mode.

### Images/gif


https://github.com/user-attachments/assets/63f79ea8-ec8f-465d-98fb-bddf2e2f762b




## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
